File: 14-longest-common-prefix/javascript/solution.js
var longestCommonPrefix = function (strs) {
  let longest = "";
  let idx = 0;
  while (strs[0] && idx < strs[0].length) {
    for (let i = 1; i < strs.length; i++) {
      if (strs[i][idx] !== strs[0][idx]) return longest;
    }
    longest += strs[0][idx];
    idx++;
  }
  return longest;
};

File: 14-longest-common-prefix/python/solution.py
class Solution(object):
    def longestCommonPrefix(self, strs):
        """
        :type strs: List[str]
        :rtype: str
        """
        if not strs:
            return ""

        for i in range(len(strs[0])):
            for str in strs[1:]:
                 if i >= len(str) or str[i] != strs[0][i]:
                    return strs[0][:i]

        return strs[0]